spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestCallStatementParser.java

```diff
@@ -188,25 +184,26 @@ private void checkArg(
 
     if (expectedName != null) {
       NamedArgument arg = checkCast(call.args().apply(index), NamedArgument.class);
-      Assert.assertEquals(expectedName, arg.name());
+      assertThat(arg.name()).isEqualTo(expectedName);
     } else {
       CallArgument arg = call.args().apply(index);
       checkCast(arg, PositionalArgument.class);
     }
 
     Expression expectedExpr = toSparkLiteral(expectedValue, expectedType);
     Expression actualExpr = call.args().apply(index).expr();
-    Assert.assertEquals("Arg types must match", expectedExpr.dataType(), actualExpr.dataType());
-    Assert.assertEquals("Arg must match", expectedExpr, actualExpr);
+    assertThat(actualExpr.dataType()).as("Arg types must match").isEqualTo(expectedExpr.dataType());
+    assertThat(actualExpr).as("Arg must match").isEqualTo(expectedExpr);
   }
 
   private Literal toSparkLiteral(Object value, DataType dataType) {
     return Literal$.MODULE$.create(value, dataType);
   }
 
   private <T> T checkCast(Object value, Class<T> expectedClass) {
-    Assert.assertTrue(
-        "Expected instance of " + expectedClass.getName(), expectedClass.isInstance(value));
+    assertThat(expectedClass.isInstance(value))
```

Comment:
```suggestion
    assertThat(expectedClass).isInstanceOf(value)
```

Reply:
the problem with the current assertion is that it only will print true/false if the assertion fails. What we rather want to achieve for assertions is to have enough context to know what actual/expected was exactly. That being said, we want to avoid the usage of `isTrue()` / `isFalse()` as much as possible.

What was the assertion that you previously used?

Reply:
Thanks for the reply. Yes, I understand the situation and we should avoid using `isTrue()`/`isFalse()`.

In the current PR, I change it to `assertThat(expectedClass.isInstance(value)).isTrue()`, 
Before submitting the PR, I tested `assertThat(expectedClass).isInstanceOf(value.getClass())`, but this failed due to the error above. This is the reason `assertThat(expectedClass.isInstance(value)).isTrue()` is used here.

Reply:
sorry my bad, the check should be `assertThat(value).isInstanceOf(expectedClass)`

Reply:
I believe `isEqualTo` can be used here. Let me change this.